### PR TITLE
Updating Slack API usage

### PR DESCRIPTION
**Pull request by marcua**
Whoda thunk?

### CODE REVIEW

File: orchestra/communication/tests/helpers/slack.py [hunk 1]
```diff
@@ -18,20 +18,20 @@ class MockSlacker(MagicMock):
...
-        self.groups = Groups()
+        self.conversations = Conversations()
         self.chat = Chat()
         self.users = Users()
         self.populate_preexisting_groups()
 
     def populate_preexisting_groups(self):
         for group_name in PREEXISTING_GROUPS:
-            self.groups.create(group_name)
+            self.conversations.create(group_name)
```
Comment (adbharadwaj) [inline orchestra/communication/tests/helpers/slack.py]: Maybe pass `is_private`?

File: orchestra/communication/slack.py [hunk 2]
```diff
@@ -69,17 +69,14 @@ def add_worker_to_project_team(worker, project):
...
         slack_user_id = worker.slack_user_id
-        response = slack.groups.invite(
+        slack.conversations.invite(
             project.slack_group_id, slack_user_id)
-        if not response.body.get('already_in_group'):
-            welcome_message = (
-                '<@{}> has been added to the team. '
-                'Welcome aboard!').format(slack_user_id)
-            slack.chat.post_message(project.slack_group_id, welcome_message)
+        welcome_message = (
```
Comment (thisisdhaas) [inline orchestra/communication/slack.py:74]: if you care I think it's `if response.body.get('error') != 'already_in_channel'`

File: orchestra/utils/tests/test_notifications.py [hunk 3]
```diff
@@ -262,7 +262,7 @@ def fake_random_string():
 
         for group_name in ('ketchup-3-sales-1', 'bongo-bash723581-3',
                            'bongo-bash723581-4'):
-            self.slack.groups.create(group_name)
+            self.slack.conversations.create(group_name)
```
Comment (adbharadwaj) [inline orchestra/utils/tests/test_notifications.py]: Pass `is_private`?

File: orchestra/communication/tests/helpers/slack.py [hunk 4]
```diff
@@ -91,25 +91,25 @@ def create(self, group_name):
...
         self._validate_user(user_id=user_id)
 
         already_in_group = True
-        if user_id not in MOCK_SLACK_API_DATA['groups'][group_id]['users']:
+        if user_id not in MOCK_SLACK_API_DATA['channels'][group_id]['users']:
             # Slacker API does not raise an error if user already present
-            MOCK_SLACK_API_DATA['groups'][group_id]['users'].append(user_id)
+            MOCK_SLACK_API_DATA['channels'][group_id]['users'].append(user_id)
             already_in_group = False
         return self.Response({'already_in_group': already_in_group})
```
Comment (thisisdhaas) [inline orchestra/communication/tests/helpers/slack.py]: change this.

File: orchestra/communication/slack.py (hunk 2 above)
Comment (marcua) [reply orchestra/communication/slack.py:74]: (Verified together that Slack/Slacker doesn't pass this through)

**Review by thisisdhaas — APPROVED** [review]

**Pull request merged**